### ISSUE TITLE
Support binarySource=install

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#11984]
### What would you like Renovate to be able to do?

Support a new option for binarySource which will use our containerbase buildpack “install-tool” commands to install languages and managers at runtime.

### If you have any ideas on how this should be implemented, please tell us here.

For now, the assumption can be that it is only enabled in a buildpack-based image which is running as `USER root`.

The biggest challenge will be syncing the datasources between Renovate and Buildpack so that we know which versions are possible (unless we use our Docker image tags themselves as the source?). We could maybe even publish an independent package which both buildpack and renovate depend on and use as their source of tool->datasource lookups?

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
I started work on this, thinking of doing this manager-by-manager with non-root.

**Comment by viceice** [comment]
@rarkins should we open a new issue to track managers which aren't ready yet?

**Comment by rarkins** [comment]
Were you thinking one common issue with checkboxes or one issue per manager? If the latter, we should open them only once buildpack supports user install.

Another big change is for me to support both language + manager dynamically. e.g. node is ready?

**Comment by viceice** [comment]
yes, that's the idea.

BTW: node should be ready for dynamic install now 🙃
